### PR TITLE
Add pop up desktop and mobile version using Javascript and DOM

### PR DESCRIPTION
**Pull request by nimplay**
This project is to dynamically add the cards using JS and at the same time create a different pop up for each card. with the purpose of improving portfolio performance.

Due to an error on my part I am using the same PR for two reviwes.
This is my second review that is about validating the email in the contact form

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -916,36 +917,48 @@ header {
...
   top: 0;
   gap: 24px;
   bottom: 0;
   background: #ffff;
   border: 1px solid #ebecf0;
-  box-shadow: 0 48px 48px rgba(37, 47, 137, 0.08);
+  box-shadow: 0 48px 48px rgb(37 47 137 / 8%);
+  margin-top: 15%;
+  margin-left: 5%;
 }
```
Comment (vechicin) [inline style.css]: Your popup windows don't quite look like those from the template design.

Your design:

Screen Shot 2022-06-15 at 8 53 04

Screen Shot 2022-06-15 at 8 53 41

The template design:

Screen Shot 2022-06-15 at 8 54 38

Screen Shot 2022-06-15 at 8 55 03

Not only is the content not right, but the way the pop-up windows are aligned is also not correct. Please fix them.

Screen Shot 2022-06-15 at 8 57 11

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @nimplay ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Good job trying to implement the pop-up window

## Required Changes ♻️

- [ ] You still have HTML content in your `Works` section. Please remove it as all of the content of that section should be dynamically generated.

Screen Shot 2022-06-15 at 9 00 18

- [ ] I don't see contributions by your coding partner. Please ask them to contribute as it is a project requirement. If you're having issues with your partner being absent please contact Student Success about it and make sure you mention that either in your request description or in the PR description.
- [ ] You are missing the PR description, please add one briefly describing what you did in this feature.

_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+{
+  title: 'Project Title 2',
+  lenguage1: 'Ruby',
+  lenguage2: 'bootstrap',
+  lenguage3: 'Ruby on Rails',
+  image: './Images/SnapshootPortfolio.png',
+  imageAlt: 'SnapshootPortfolio',
+  description: 'Second Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consequatur amet quasi suscipit enim quidem quas odio libero totam nisi. Numquam corrupti sequi eligendi error laboriosam quasi nulla quidem eaque reiciendis.',
+}];
+/* Functions */
```
Comment (danmainah) [inline app.js]: - [ ] Kindly ensure all your projects are in the array and are displayed on your page. Currently, you only have 1 project.
- [ ]  Kindly check the design of your popup, the Figma design has the background blurred and a bit dark kindly check on it here is the [link](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=34%3A1278) to the design.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @nimplay and @rominana, 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


 # Required Changes ♻️

- [ ] Kindly add a pull request summary your project is missing a pull request summary.
image


_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nimplay** [comment]
Summary:
I eliminated the cars as the reviwer said and added them dynamically in the pop up, I also created the carousel to be able to see the 5 cards and added the figma text. I have also created an animation for the call to action buttons. I hope this time is enough to aprove.

File: app.js [hunk 3]
```diff
@@ -6,4 +67,80 @@ function toggleMenu() {
...
+        <a class="pop-up-button" href="https://github.com/nimplay/Microverse_Portfolio">
+          <p>See Source</p>
+          <img src="./Images/whiteGitHub.svg" alt="">
+        </a>              
+      </div>
+    </div>
+    <div class="pop-up-arrow" id="arrow2"></div>
+  </div>
+  </div>
+`;
```
Comment (youmari) [inline app.js]: - [x] please make sure that you displayed the cards of the project as in the Figma design right now I see just one 
image

File: app.js [hunk 4]
```diff
@@ -6,4 +67,80 @@ function toggleMenu() {
...
+        </a>
+        <a class="pop-up-button" href="https://github.com/nimplay/Microverse_Portfolio">
+          <p>See Source</p>
+          <img src="./Images/whiteGitHub.svg" alt="">
+        </a>              
+      </div>
+    </div>
+    <div class="pop-up-arrow" id="arrow2"></div>
+  </div>
+  </div>
```
Comment (youmari) [inline app.js]: - [x] [optional] the image of the projects cards is overflowing above the content please try to fix that 
image

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @nimplay , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Great title
✔️ Github flow 
✔️ Good commit messages
✔️ No linter error

 ## Required Changes ♻️

- [ ]  please try to add the  short summary  that tells in more detail what you've done  in the PR
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js [hunk 5]
```diff
@@ -6,4 +67,80 @@ function toggleMenu() {
...
+        <a class="pop-up-button" href="https://nimplay.github.io/Microverse_Portfolio/#">
+          <p>See Live</p>
+          <img src="./Images/liveServer.svg" alt="">
+        </a>
+        <a class="pop-up-button" href="https://github.com/nimplay/Microverse_Portfolio">
+          <p>See Source</p>
+          <img src="./Images/whiteGitHub.svg" alt="">
+        </a>              
+      </div>
+    </div>
```
Comment (AmaduKamara) [inline app.js]: - [ ] I can see that you are manually displaying only one card. The idea here is that for the array of projects you already created, you need to loop through that array and render the lopped data in the section where you're manually displaying a single project. That way, all six projects will get displayed on the screen just as in the Figma template. 
- [ ] For this project you are not required to make the project's modal displayed in a carousel fashion. You are required to have six projects displayed on the screen and when someone clicks on a single project to view its details, the details for that project are then displayed on the popup window. Please use that method in order for your project to be completed. I trust you can 👍 
- [ ] Also, please work on the image that is overlapping the badges on your card or cards(when you must have implemented the lopped cards). Please work on your CSS styles to correctly align card items on your project.

**See your Project**
image

**The Figma Template**
image

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @rominana,

AlonzoLeroneThisCloseGIF (2)

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ Correct use of Github workflow
✅ No linter error
✅ README is okay

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly work on the changes asked by previous reviewers and mostly follow through with the summary guide I required. It is best to follow project requirements and instructions for every task you are given. It's going to be the same in your future job and hence, keep the practice going 👍 
- [ ] Kindly add a Pull Request description to your PR. 
- [ ] Click on the three dots I circled and then edit, in the description box that opens, kindly write all the tasks you worked on in this project with your partner. You should be doing that for all future projects 👍 
image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by nimplay** [comment]
sumary:
I have dynamically added the cards and at the same time created a different pop up for each card, at the same time I fixed the details of the format and solved the linters errors according to the requirements of the project. I hope this time it will be approved

**Review by Kandy-Peter — APPROVED** [review]
Hi @nimplay ,

# APPROVED 🟢 


Great implementation of all the requirements in this milestone. 👏
Your web app looks really fantastic. 🎉 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

#### CONGRATULATIONS! 🎉 

##  Optional suggestion

- [ ] OPTIONAL: It would be better if you align your cards to the center to stick the Figma design.
**1.Your design** 
image

**2.Figma design**

image

# CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nimplay** [comment]
giphy

SUMMARY:
The functionalities were added using javascript so that the contact form would recognize if the text is in lowercase. in addition to changing the background and displaying an error message

**Review by smunozmo — APPROVED** [review]
Hi @nimplay, 👋 

## Highlights 👏 

- Form validation is working properly, nice job!
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

### APPROVED! 💯  💯  💯 




Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**